server/src/main/java/org/opensearch/discovery/PluginRequest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.discovery;
+
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.extensions.DiscoveryExtensionNode;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * PluginRequest to intialize plugin
```

Comment:
nit: Rename Plugin to Extension? 
```suggestion
 * ExtensionRequest to initialize extension
```

Reply:
That should be accomplished in the next PR, but I can add it now if you want.

Reply:
Eventually, it will be InitializeExtensionsRequest based on the current feature/extensions branch